library_management_system_downloader/copy_and_compress_database_and_extract_tsv.py

```diff
@@ -0,0 +1,102 @@
+# =============================================================================
+# Import libraries
+# =============================================================================
+
+import logging
+import lzma
+import os
+import pandas as pd
+from sqlalchemy import create_engine
+from sqlalchemy.ext.automap import automap_base
+from sqlalchemy.orm import sessionmaker
+
+# =============================================================================
+# Settings
+# =============================================================================
+
+sqlite_database_location = 'library_coverage_xml_and_fulltext_indicators.db'
+
+data_directory = 'data'
+
+compressed_database_copy_name = (
+        'library_coverage_xml_and_fulltext_indicators'
+        '.db.xz')
+
+tsv_output_file_name = 'library_coverage_xml_and_fulltext_indicators.tsv.xz'
+
+# =============================================================================
+# Create additional settings
+# =============================================================================
+
+logging.basicConfig(level=logging.INFO)
+
+# =============================================================================
+# Set up a connection to (and tables within) the SQLite database:
+# =============================================================================
+
+# Initialize the database:
+Base = automap_base()
+sql_engine = create_engine(f'sqlite:///{sqlite_database_location}')
+Base.prepare(sql_engine, reflect=True)
+
+dois_table = Base.classes.dois_table
+library_holdings_data = Base.classes.library_holdings_data
+
+# From the sqlalchemy documentation
+# (http://docs.sqlalchemy.org/en/latest/orm/session_basics.html)
+# Create a configured "session":
+Session = sessionmaker(bind=sql_engine)
+sql_session = Session()
+
+# =============================================================================
+# Create the data directory if it doesn't already exist
+# =============================================================================
+
+if not os.path.exists(data_directory):
+    os.makedirs(data_directory)
```

Comment:
Not essential but you may enjoy `pathlib`, which is now builtin and is great for doing things with paths.

Reply:
Thanks! Implemented in fb053e5.